nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/ConsumerPool.java

```diff
@@ -416,7 +407,7 @@ protected String generateReaderId() {
      */
     protected EventStreamReader<byte[]> createPravegaReader(final String readerId) {
         logger.debug("createPravegaReader: readerId={}", new Object[]{readerId});
-        final EventStreamReader<byte[]> reader = clientFactory.createReader(
+        final EventStreamReader<byte[]> reader = EventStreamClientFactory.withScope(scope, clientConfig).createReader(
```

Comment:
Having removed `clientFactory` with the removal of `batchClient`, a cached factory is no longer use here. I wasn't sure if that would be an issue or not.